File: 102/1912.py
n = int(input())
arr = list(map(int,input().split()))
answer = -1001
total = 0
for v in arr:
    total += v
    answer = max(answer,total)
    if total < 0:
        total = 0
print(answer)
# 백준 연속 합 문제
# 음수를 더한다고 해서 항상 손해는 아니기에 더해보고, 음수가 나오면 0으로 다시 시작.

File: 1103/1912.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const solution = () => {
  const dp = new Array(n).fill(-1001); // 가장 최소가 되는 값은 -1001이므로 이를 할당 해준다.
  dp[0] = arr[0];
  for (let i = 1; i < n; i++) {
    dp[i] = Math.max(dp[i], dp[i - 1] + arr[i]);
  }
  console.log(Math.max(...dp));
};
let n = null;
let arr = null;
rl.on("line", (line) => {
  if (!n) {
    n = +line;
  } else {
    arr = line.split(" ").map((e) => +e);
    rl.close();
  }
});
rl.on("close", () => {
  solution();
  process.exit();
});
//백준 연속합 문제
